### ISSUE TITLE
Better Language Indicator

### ISSUE THREAD

**Issue by patrickdark** [callahad/quick-accept-language-switcher#7]
If I enter something like "eng;q=0,zho", this disallows English, but "eng" is nevertheless displayed over the icon when I'd expect "zho".

Seems like it'd be better to either determine the highest weighted languages and display the code for the highest weighted language or "mul" (an ISO 639-2 language code indicating multiple languages) if there are multiple highest weighted languages.

Alternatively, one could do no weighting at all and display something indicating multiple languages.

Also, I'm not sure if invalid input like "???" should be displaying anything. (It displays "???".)

**Comment by callahad** [comment]
Good idea! The badge API can only display around four characters, so right now it's just showing [the beginning of the string](https://github.com/callahad/quick-accept-language-switcher/blob/597222de5008d7c33571af340b020cb08cc0185a/popup.js#L6-L9), mainly as a reminder that *something* is set, but you're right that showing the locale with the highest qvalue would be more informative.

Are you interested in fixing this? I'm totally open to pull requests. :)

**Comment by patrickdark** [comment]
@callahad Yes, I can fix this if I can figure out how a pull request works. (I've written my own extensions.)

It looks like I'm supposed for fork the project, update popup.js, and merge it back into this project? I don't understand how one merges modified code back into the project though.

**Comment by callahad** [comment]
Awesome! No time like the present to learn how GitHub handles this stuff. :) Are you familiar with `git` itself, or are you also learning that at the same time?

You've got the right idea on the workflow:

1. Fork this repository
2. Edit your copy
3. Propose changes (pull request)

Once you've updated popup.js, GitHub should show you a big yellow banner with a button to create a pull request. When you do that, I'll get a notification and be able to review and incorporate your changes.

If anything doesn't make sense, let me know. I'm happy to help, and even happier for this project to gain its first external contributor!

**Comment by patrickdark** [comment]
@callahad I downloaded Git Bash and Visual Studio 2015 and created some special command line ID when I tried to contribute a Firefox patch for an issue that required trivial JavaScript changes to their codebase, but then gave up since I couldn't figure it out.

I'm familiar with GitHub in a basic sense as I regularly submit bug reports  and `git` as in I know it's a command line item, but that's about it. I don't know how to pull code from a specific GitHub repository, for example, so I just manually downloaded every file from my forked copy of yours.

I created a pull request with tested changes at https://github.com/callahad/quick-accept-language-switcher/pull/8. It's a total file overwrite so the diff page doesn't look nice. Not sure how to do it correctly.

I didn't see a yellow banner. I uploaded a file to the master, clicked a white "New pull request" button on the fork's main page then clicked a green pull request creation button afterward.

And glad to help!

**Comment by patrickdark** [comment]
So apparently I get the nice diff by using GitHub's file editor instead of doing a whole file overwrite.

**Comment by callahad** [comment]
Overwriting the whole file should be fine -- git should be able to generate its own, minimal diff between the two versions -- but somehow that didn't happen this time. Weird. :)

It might be worth checking out Microsoft's lighter-weight [VSCode](https://code.visualstudio.com) editor, which has a nice [built-in Git client](https://code.visualstudio.com/Docs/editor/versioncontrol) that should make it easier to see exactly what is being committed. I presume VS2015 has something similar, but I haven't used it.
